### ISSUE TITLE
make check fails on .tar.gz release

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/BOUT-dev#1364]
The tests try to clone the submodule, which fails
```
echo "Downloading Google Test"
Downloading Google Test
git submodule update --init --recursive
fatal: not a git repository (or any of the parent directories): .git
make[1]: *** [makefile:46: ../../googletest/README.md] Error 128
make: *** [makefile:61: build-check-unit-tests] Error 2
```

**Comment by d7919** [comment]
Just to clarify -- do you mean when downloading from https://github.com/boutproject/BOUT-dev/releases/tag/v4.2.0 ?

**Comment by dschwoerer** [comment]
This one:
https://github.com/boutproject/BOUT-dev/archive/v4.2.0.tar.gz

It you download via git clone, then everything is fine (I assume) but via the tar.gz, git submodule does not work to download googletest.

An option would be to make this test optional - e.g. skip if the download fails.

**Comment by d7919** [comment]
Any ideas @ZedThree ?

**Comment by ZedThree** [comment]
It seems GitHub doesn't include submodules as part of the release (as they might point to private repos). I think some makefile magic is needed. `git rev-parse --is-inside-work-tree` will return `true` if in a git repo, and some error if not. The recipe for `$(GTEST_SENTINEL)` could then check that somehow and `git clone` google test if it's not in a git repo.

I'll see if I can bash something together now

**Comment by dschwoerer** [comment]
Note that that condition may be true, but it may still fail, e.g. if:
 * Bout is within another repository
 * No internet

**Comment by ZedThree** [comment]
If you have no internet, there's nothing we can do. Likewise if GitHub is down.

If BOUT++ is in another repo, I'm not sure there's anything we can do in that case either. That would be very hard to check for. Best hope whoever did that knows what they are doing!

I think we can only try so hard to recover from bad situations. I have something that works and bails if it can't handle the situation, it is hopefully enough for the majority of situations

**Comment by dschwoerer** [comment]
So you want to download it some other way?
I thought the tests would be skipped in that case, but downloading is
probably better for the average case.

You could check that .git exists in the BOUT_TOP - that should tell you
whether bout is under git version control. Another option would be to
check whether a specific commit (full hash) is in the history.

**Comment by dschwoerer** [comment]
Related https://github.com/dschwoerer/BOUT-dev/issues/11

The release creation script should ideally also generate the boutcore cpp files, and provide a makefile that works without cython.

Further, all cxx files should have a newer timestamp than the source files, to not require jinja2.

**Comment by ZedThree** [comment]
Fixed in #1482, will upload separate tarballs with a health warning for the GitHub provided ones
